doc/project/checklist.md

```diff
@@ -395,3 +395,91 @@ Clean up all compile-time and run-time warnings thrown by the simulator.
 ### PRE_VERIFIED_SUB_MODULES_V3
 
 Sub-modules that are pre-verified with their own testbenches have already reached V3 stage.
+
+## S1
+
+For a transition from S0 to S1, the following items are expected be completed.
+
+### DIF_EXISTS
+
+`dif_<ip>.h` and, optionally, `dif_<ip>.c` exist in `sw/device/lib/dif`.
+
+### DIF_USED_IN_TREE
+
+All existing in-tree code which uses the device, uses the device via the DIF.
+There is no remaining driver code that directly uses the device outside of DIF code.
+
+### DIF_TEST_UNIT
+
+Unit tests exist for the DIF in `sw/device/tests/dif` named `dif_<ip>_unittest.cc`.
+
+### DIF_TEST_SANITY
+
+Sanity tests exist for the DIF in `sw/device/tests/dif` named `dif_<ip>_sanitytest.c`.
+This should perform an end-to-end use of the hardware via the DIF, and should be able to be run on all OpenTitan platforms (including FPGA and simulation).
+
+## S2
+
+For a transition from S1 to S2, the following items are expected be completed.
+
+### DIF_FEATURES
+
+DIF has functions to cover all specified hardware functionality.
+
+### DIF_HW_FEATURE_COMPLETE
+
+The DIF's respective device IP is at least stage D2.
+
+### DIF_HW_PARAMS
+
+DIF uses automatically generated HW parameters and register definitions.
```

Comment:
I see, okay. Wouldn't the reggen / topgen stabilize at some point (probably very soon) once those extra definitions get added for auto-generation, so future DIFs wont have to hand-write anything at all? Just curious, fine with keeping this here in any case.

Reply:
Yes, we do envision reggen stabilising eventually.

I realise actually that topgen is less of an issue, as beyond parameterisations of peripherals (which are captured by reggen running on a topgen'd `.hjson` at the moment), because we're trying not to have any device-specific code in DIFs themselves. In effect, topgen is responsible for generating information that a user needs to use a DIF, but nothing internal to the DIF's implementation.